components/core/tests/test-ir_encoding_methods.cpp

```diff
@@ -256,32 +264,46 @@ TEMPLATE_TEST_CASE("decode_preamble", "[ffi][decode_preamble]", four_byte_encode
     const size_t encoded_preamble_end_pos = ir_buf.size();
 
     // Check if encoding type is properly read
-    IrBuffer preamble_buffer(ir_buf.data(), ir_buf.size());
+    BufferReader encoding_buffer{
+            size_checked_pointer_cast<const char>(ir_buf.data()), ir_buf.size()
+    };
     bool is_four_bytes_encoding;
-    REQUIRE(get_encoding_type(preamble_buffer, is_four_bytes_encoding) ==
+    REQUIRE(get_encoding_type(encoding_buffer, is_four_bytes_encoding) ==
             IRErrorCode::IRErrorCode_Success);
     REQUIRE(match_encoding_type<TestType>(is_four_bytes_encoding));
-    REQUIRE(MagicNumberLength == preamble_buffer.get_cursor_pos());
+    REQUIRE(MagicNumberLength == encoding_buffer.get_pos());
 
     // Test if preamble can be decoded correctly
     TimestampInfo ts_info;
-    int8_t metadata_type{0};
+    encoded_tag_t metadata_type{0};
     size_t metadata_pos{0};
     uint16_t metadata_size{0};
-    REQUIRE(decode_preamble(preamble_buffer, metadata_type, metadata_pos, metadata_size) ==
+    REQUIRE(decode_preamble(encoding_buffer, metadata_type, metadata_pos, metadata_size) ==
+            IRErrorCode::IRErrorCode_Success);
+    REQUIRE(encoded_preamble_end_pos == encoding_buffer.get_pos());
+    
+    auto json_metadata_ptr = reinterpret_cast<char*>(ir_buf.data() + metadata_pos);
+    string_view json_metadata_ref {json_metadata_ptr, metadata_size};
+
+    // Test if preamble can be decoded by the string copy method
```

Comment:
I think this would be clearer if you finished testing the first `decode_preamble` method and then tested the second one.